File: src/Services/WareHouse/WareHouse.API/Application/Model/WareHouseItemUnitDTO.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace WareHouse.API.Application.Model
{
    public partial class WareHouseItemUnitDTO: BaseModel
    {
        public string ItemId { get; set; }
        public string UnitId { get; set; }
        public string UnitName { get; set; }
        public int ConvertRate { get; set; }
        public bool? IsPrimary { get; set; }

        public virtual WareHouseItemDTO Item { get; set; }
        public virtual UnitDTO Unit { get; set; }
    }
}

File: src/WebClient/WareHouse/src/app/model/WareHouseItemUnitDTO.ts
import { BaseModel } from "./BaseModel";
import { UnitDTO } from "./UnitDTO";
import { WareHouseItemDTO } from "./WareHouseItemDTO";

export interface WareHouseItemUnitDTO extends BaseModel {
    itemId: string;
    unitId: string;
    unitName: string;
    convertRate: number;
    isPrimary: boolean | null;
    item: WareHouseItemDTO| null;
    unit: UnitDTO| null;
}